### PR TITLE
Get tests passing on Rails 5

### PR DESCRIPTION
**Pull request by bf4**
Having trouble with caching tests.  Help?

### CODE REVIEW

**Comment by bf4** [comment]
Failures for Rails 5 are in caching. Help, anyone?

ref:

- https://github.com/rails-api/active_model_serializers/issues/1045
- https://github.com/rails-api/active_model_serializers/issues/1020
- https://github.com/rails-api/active_model_serializers/issues/827
- https://github.com/rails-api/active_model_serializers/pull/810

cc @kurko @guilleiguaran @joaomdmoura @spastorino @thibaudgg

**Comment by bf4** [comment]
ref: https://github.com/rails-api/active_model_serializers/issues/1303

File: test/action_controller/serialization_scope_name_test.rb [hunk 1]
```diff
@@ -12,7 +12,12 @@ def admin?
   class UserTestController < ActionController::Base
     protect_from_forgery
 
-    before_filter { request.format = :json }
+    # For Rails5
+    if self.respond_to?(:before_action)
+      before_action { request.format = :json }
+    else
+      before_filter { request.format = :json }
+    end
```
Comment (bf4) [inline test/action_controller/serialization_scope_name_test.rb]: yup

Comment (beauby) [inline test/action_controller/serialization_scope_name_test.rb]: I believe `before_action` already exists in rails4.

Comment (bf4) [inline test/action_controller/serialization_scope_name_test.rb]: yeah, but not in all versions

**Pull request merged**